api/src/main/kotlin/middleware/ApiKeyMiddleware.kt

```diff
@@ -0,0 +1,38 @@
+package com.wafflestudio.snu4t.middleware
+
+import com.wafflestudio.snu4t.common.exception.WrongApiKeyException
+import com.wafflestudio.snu4t.handler.RequestContext
+import io.jsonwebtoken.Jwts
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.stereotype.Component
+import org.springframework.web.reactive.function.server.ServerRequest
+import javax.crypto.spec.SecretKeySpec
+
+@Component
+class ApiKeyMiddleware(
+    @Value("\${snutt.secret-key}") private val secretKey: String,
+) : Middleware {
+    private val keySpec = SecretKeySpec(secretKey.toByteArray(), "HmacSHA256")
+    private val jwtParser = Jwts.parser().setSigningKey(keySpec)
+    private val stringToKeyVersionMap = mapOf(
+        "ios" to "0",
+        "web" to "0",
+        "android" to "0",
+        "test" to "0",
+    )
+
+    override suspend fun invoke(req: ServerRequest, context: RequestContext): RequestContext {
+        try {
+            val apiKey = requireNotNull(req.headers().firstHeader("x-access-apikey"))
+            val claims = jwtParser.parseClaimsJws(apiKey).body
+
+            val string = claims["string"]!!.toString()
+            val keyVersion = claims["key_version"]!!.toString()
+            require(stringToKeyVersionMap[string] == keyVersion)
+        } catch (e: Exception) {
+            throw WrongApiKeyException
+        }
+
+        return context
```

Comment:
없애자고 제안한 마당에 이상하지만

```suggestion
    override suspend fun invoke(req: ServerRequest, context: RequestContext): RequestContext =
        runCatching {
            val apiKey = requireNotNull(req.headers().firstHeader("x-access-apikey"))
            val claims = jwtParser.parseClaimsJws(apiKey).body

            val string = claims["string"]!!.toString()
            val keyVersion = claims["key_version"]!!.toString()
            require(stringToKeyVersionMap[string] == keyVersion)
            context
        }.getOrElse { throw WrongApiKeyException }
```
이거 어떠신가요ㅋㅋ

Reply:
반영하겠습니다. `context` 만 `return@runCatching context`으로 할게요.